build-system/tasks/helpers.js

```diff
@@ -847,6 +818,35 @@ function mkdirSync(path) {
   }
 }
 
+/**
+ * Returns the list of dependencies for a given JS entrypoint.
+ * @param {string} entryPoint
+ * @return {Promise<Array<string>>}
+ */
+async function getDependencies(entryPoint) {
+  const stripImportAssertionPlugin = {
+    name: 'stripImportAssertion',
+    setup(build) {
+      build.onLoad({filter: /\.[cm]?js$/, namespace: ''}, async (file) => {
+        const filename = file.path;
+        const contents = (await batchedRead(filename)).contents.toString();
+        const stripped = contents.replace(/assert {type: 'json'}/g, '');
+        return {contents: stripped};
+      });
+    },
+  };
+
+  const result = await esbuild.build({
+    entryPoints: [entryPoint],
+    bundle: true,
+    write: false,
+    metafile: true,
+    plugins: [stripImportAssertionPlugin],
+  });
+
+  return Object.keys(result.metafile?.inputs);
```

Comment:
Just checking, will this work for files with zero imports?

Reply:
Object.keys should return an empty list, so I assume so